cli/src/cmd/cast/zk_deposit.rs

```diff
@@ -110,37 +130,33 @@ impl ZkDepositTxArgs {
     /// - Ok: If the deposit transaction is successfully completed.
     /// - Err: If an error occurred during the execution of the deposit transaction.
     pub async fn run(self) -> eyre::Result<()> {
-        let private_key = get_private_key(&self.wallet.private_key)?;
+        let private_key = &self.wallet.private_key.as_ref().unwrap();
         let l1_url = get_rpc_url(&self.l1_url)?;
         let l2_url = get_url_with_port(&self.l2_url).expect("Invalid L2_RPC_URL");
         let chain = get_chain(self.chain)?;
-        let signer = get_signer(private_key, &chain);
-        let wallet = wallet::Wallet::with_http_client(&l2_url, signer);
-        let to_address = self.get_to_address();
-        let token_address: Address = match self.token {
-            Some(token_addy) => token_addy,
-            None => Address::zero(),
-        };
+        // let token_address: Address = match self.token {
+        //     Some(token_addy) => token_addy,
+        //     None => Address::zero(),
+        // };
 
-        match wallet {
-            Ok(w) => {
-                println!("Bridging assets....");
-                let eth_provider = w.ethereum(l1_url).await.map_err(|e| e)?;
-                let tx_hash = eth_provider
-                    .deposit(
-                        token_address,
-                        self.amount,
-                        to_address,
-                        self.operator_tip,
-                        self.bridge_address,
-                        None,
-                    )
-                    .await?;
-
-                println!("Transaction Hash: {:#?}", tx_hash);
-            }
-            Err(e) => eyre::bail!("Failed to download the file: {}", e),
-        }
+        let l1_provider = Provider::try_from(l1_url)?;
+        let l2_provider = Provider::try_from(l2_url)?;
+        let wallet = LocalWallet::from_str(private_key).unwrap().with_chain_id(chain.id());
+        let zk_wallet =
+            ZKSWallet::new(wallet, None, Some(l2_provider.clone()), Some(l1_provider.clone()))
+                .unwrap();
+
+        let deposit_request = DepositRequest::new(self.amount.into())
+            .to(self.get_to_address())
+            .operator_tip(self.operator_tip)
+            .gas_price(self.tx.gas_price)
```

Comment:
ok, now I understand why you added the `Option` to the `DepositRequest`. I still think it would be better for these functions to not receive an Option

Reply:
Fixed